experiment/src/cortex/experiment/classification.clj

```diff
@@ -188,18 +192,26 @@
                                     (count test-ds)))
         best-network? (or (nil? old-classification-accuracy)
                           (> (double classification-accuracy)
-                             (double old-classification-accuracy)))]
+                             (double old-classification-accuracy)))
+        updated-network (if best-network?
+                          (let [best-network
+                            (assoc new-network 
+                              :classification-accuracy classification-accuracy)]
+                            (reset-confusion-matrix confusion-matrix-atom
+                                                    observation->img-fn
+                                                    class-mapping
+                                                    {:labels labels
+                                                    :test-ds test-ds})
+                            (experiment-train/save-network best-network network-filename))
+                            ;;seems dicey. if not the best-network, 
```

Comment:
You're totally right.

For this simple example, this performs well. But moving forward we'll be exposing more of our sophisticated training regimes as appropriate.